cmd/object-handlers.go

```diff
@@ -1244,6 +1247,9 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 			return
 		}
 	}
+	if globalAutoEncryption && !crypto.SSEC.IsRequested(r.Header) {
+		r.Header.Add(crypto.SSEHeader, crypto.SSEAlgorithmAES256)
+	}
```

Comment:
move this check after line 1254 - auto encryption headers needed only if encryption is supported by object layer

Reply:
See: https://github.com/minio/minio/pull/6523#discussion_r241499044

Reply:
@aead, agree - missed that globalAutoEncryption being set automatically implies object layer supports encryption, since you are checking it at startup.